client/app/pages/queries-list/QueriesList.jsx

```diff
@@ -86,6 +87,22 @@ class QueriesList extends React.Component {
     }),
   ];
 
+  componentDidMount() {
```

Comment:
Shouldn't the component be re-rendered when the URL updates?

Reply:
Well, no. Router will recreate controller only if path changes. Watching for url parameters is responsibility of each controller

Reply:
This is not how I would expect a router to work in React and in general it leads to repeating this logic anywhere where we might be using query string parameters. 

I would've expected the page to receive the parameters as props, which would trigger React's rendering when/if needed.

Was there some explicit reason to make it work this way?

Reply:
(this is a general discussion, not something necessary to address in this PR)

Reply:
> Was there some explicit reason to make it work this way?

That was a sort of optimization - to make sure that router will reload controllers only when needed. At that point, it seemed a good idea to bind controllers only to URL path so if path doesn't change - the same controller is kept. But now I agree with your point and I think we need to revisit this mechanism. It will require some changes to `ItemsList` component as I guess only controllers based on it use URL parameters. I'll create an issue for this under a `tech debt` label

Reply:
Issue getredash/redash#4864